conans/test/functional/shims/test_shims.py

```diff
@@ -0,0 +1,172 @@
+            std::cout << "library-envvar: " << env_variable << "\\n";
+        }
+    """)
+
+    lib_h = textwrap.dedent("""
+        #pragma once
+
+        void message();
+    """)
+
+    lib_conanfile = textwrap.dedent("""
+        from conans import ConanFile, CMake
+
+        class Recipe(ConanFile):
+            name = 'library'
+            options = {'shared': [True, False]}
+            default_options = {'shared': True}
+            generators = 'cmake'
+            exports_sources = "*"
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.definitions['LIBRARY_VERSION'] = self.version
+                cmake.configure()
+                cmake.build()
+                cmake.install()
+
+            def package_info(self):
+                self.cpp_info.libs = ['library']
+                self.env_info.LIBRARY_ENVVAR = 'library-{}'.format(self.version)
+    """)
+
+    lib_cmakelists = textwrap.dedent("""
+        cmake_minimum_required(VERSION 2.8.12)
+        project(library CXX)
+
+        include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+        conan_basic_setup()
+
+        set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+        add_library(${PROJECT_NAME} library.cpp)
+        set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER library.h)
+        target_compile_definitions(${PROJECT_NAME} PRIVATE LIBRARY_VERSION="${LIBRARY_VERSION}")
+        install(TARGETS ${PROJECT_NAME}
+            RUNTIME DESTINATION bin
+            LIBRARY DESTINATION lib
+            ARCHIVE DESTINATION lib
+            PUBLIC_HEADER DESTINATION include)
+    """)
+
+    # Files and recipe for two build-requires that will consume previous library:
+    main_cpp = textwrap.dedent("""
+        #include "library.h"
+
+        int main() {
+            message();
+        }
+    """)
+
+    main_cmakelist = Template(textwrap.dedent("""
+        cmake_minimum_required(VERSION 2.8.12)
+        project({{ project }} CXX)
+
+        include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+        conan_basic_setup(TARGETS)
+
+        add_executable(${PROJECT_NAME} main.cpp)
+        target_link_libraries(${PROJECT_NAME} CONAN_PKG::library)
+        install(TARGETS ${PROJECT_NAME} RUNTIME DESTINATION bin)
+    """))
+
+    main_conanfile = Template(textwrap.dedent("""
+        from conans import ConanFile, CMake
+
+        class Recipe(ConanFile):
+            name = '{{ project }}'
+            requires = 'library/{{ lib_version }}'
+            exports = "*"
+            generators = 'cmake'
+
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure()
+                cmake.build()
+                cmake.install()
+
+            def package_info(self):
+                self.cpp_info.exes = ["{{ project }}"]
+                self.env_info.LIBRARY_ENVVAR = '{{ project }}-value'
+                #self.env_info.PATH = [os.path.join(self.package_folder, "bin")]
```

Comment:
Using _shims|wrappers_ it is not needed to define/add this folder to the `PATH`.